matcher.go

```diff
@@ -1,43 +1,61 @@
 package gostatsd
 
 import (
+	"regexp"
 	"strings"
 )
 
 type StringMatch struct {
```

Comment:
I wonder if it'd be better to turn this into an interface with multiple simple implementations. Perhaps it's not worth the effort at this point, just a thought.

Reply:
I'd been thinking that. Definitely something to benchmark and not do blindly, as this is one of the hottest code paths in the service.